src/main/kotlin/net/ccbluex/liquidbounce/config/types/Value.kt

```diff
@@ -362,6 +363,76 @@ class BindValue(
     }
 }
 
+class MultiChooseListValue<T>(
+    name: String,
+    value: Array<T>,
+
+    @Exclude
+    val choices: Array<T>,
+
+    @Exclude
+    @ProtocolExclude
+    private val clazz: KClass<T>
+) : Value<Array<T>>(
+    name,
+    defaultValue = value,
+    valueType = ValueType.MULTI_CHOOSE,
+    listType = ListValueType.Enums
+) where T : Enum<T>, T : NamedChoice {
+    /**
+     * For optimization purposes.
+     *
+     * O(n) -> O(1)
+     */
+    private val indexMap = choices.distinct().mapIndexed { index, choice -> choice to index }.toMap()
+
+    override fun deserializeFrom(gson: Gson, element: JsonElement) {
+        val active = mutableSetOf<T>()
+
+        if (element.isJsonArray) {
+            for (item in element.asJsonArray) {
+                choices.firstOrNull { it.choiceName == item.asString }?.let {
+                    active.add(it)
+                }
+            }
+        }
+
+        active.sortedBy {
+            indexMap[it] ?: Int.MAX_VALUE
+        }.toTypedArray().let {
+            set(it)
+        }
+    }
+
+    fun toggle(value: T): Boolean {
+        val current = get().toMutableSet()
+
+        if (value in current) {
+            current.remove(value)
+        } else {
+            current.add(value)
+        }
+
+        current.sortedBy {
+            indexMap[it] ?: Int.MAX_VALUE
+        }.toTypedArray().let {
+            set(it)
+        }
+
+        return value in current
+    }
+
+    @Suppress("NOTHING_TO_INLINE")
+    inline operator fun contains(choice: T) = get().contains(choice)
```

Comment:
kotlinc even tells you that this inline operator is a bad idea. Don't use it if you don't need it, there is a JIT compiler.